Sources/PackageModelSyntax/AddTarget.swift

```diff
@@ -414,3 +414,15 @@ fileprivate extension PackageDependency {
         )
     }
 }
+
+fileprivate extension TargetDescription {
+    var sanitizedName: String {
+        name
+            .spm_mangledToC99ExtendedIdentifier()
+            .capitalizingFirstLetter()
+    }
+}
+
+fileprivate extension String {
+    func capitalizingFirstLetter() -> String { prefix(1).uppercased() + dropFirst() }
```

Comment:
Using both `prefix(1)` and `dropFirst()` is absolutely safe with strings of any length, including empty ones, as both are concerned with a maximum number of characters to process, not a minimum.

Reply:
I've decided to introduce `capitalizingFirstLetter()` instead of using the existing `capitalized()` function because the latter has an undesirable by-design behavior. It lowercases all remaining letters in words, so "MyNewShinyTarget" would be turned into "Mynewshinytarget". This makes it a rather poor option for snake-case notation, which is the most popular convention for target names.

On the other hand, capitalizing only the first letter would turn "my-new-shiny-target" into "My_new_shiny_target" instead of the potentially preferred "My_New_Shiny_Target" (which `capitalized()` would produce), but using underscores would be a non-preferred Swift naming convention anyway. Therefore I would consider it a very minor downside given that we're talking about example names here.

Reply:
So, worth taking a moment to write some unit tests where the macro name does _not_ use the English Latin alphabet. For instance, does the French "être" correctly capitalize to "Être"? Does the Japanese "マクロ" stay unmodified? These are trivial test cases but worth adding.

More complex to solve would be something like the "Turkish I" capitalization problem (uppercase "i" in Turkish is "İ", not "I", and lowercase "I" is "ı".) Should this code attempt to take the user's localization settings into consideration?

Reply:
Thanks, that's a good point again. I switched to `localizedCapitalized` instead of `uppercased()`, which addresses all the cases mentioned above while taking the current user's locale into account. Interestingly, the case with the Turkish "İ" is [specifically covered in Foundation](https://github.com/apple/swift-foundation/blob/8bf4cb0b9be0cb12ce87d85d2ceafaa605584051/Tests/FoundationEssentialsTests/StringTests.swift#L1510-L1512) :) I'm surprised that the tests are far from being rigorous...

I'd love to write more tests that cover localization edge cases. However, this would require some refactoring of the test helpers (as I [mentioned here](https://github.com/swiftlang/swift-package-manager/pull/7764#discussion_r1668393979)) if we want to avoid significant duplication in tests. 
An alternative approach could be to use the Turkish "I" in the existing test target names to confirm that at least some localization is taken into account. However, I haven't found a way to properly mock `Locale.current` in tests.

At the same time, what we have might already be good enough for this specific use case, as the main goal of this PR was to prevent broken code in modified packages in the first place.

@DougGregor I'd also like to hear your opinion.

Reply:
I just checked, and `Locale.current` correctly reports `en_DE` in my case, which is aligned with the system settings:
```
defaults read ~/Library/Preferences/.GlobalPreferences AppleLocale
en_DE
```
At the same, the system `locale` utility reports the default `en_US`, as you said:
```
locale
LANG="en_US.UTF-8"
```
Meaning, `String.localizedCapitalized` should work as expected according to the user's localization preferences.